packages/block-editor/src/hooks/align.js

```diff
@@ -138,7 +138,18 @@ function BlockEditAlignmentToolbarControls( {
 				nextAlign = '';
 			}
 		}
-		setAttributes( { align: nextAlign } );
+		/**
+		 * This filter exists so that the image block can reset
+		 * its aspect ratio and custom size when the alignment is set
+		 * to either wide or full.
+		 */
+		const filteredAttributes = applyFilters(
+			'block-library.image.alignmentUpdate',
```

Comment:
It feels weird to me to add image block specific code to the align hook. Not sure if adding a filter here is the right way to go. I like @fabiankaegy's [suggestion to add an effect](https://github.com/WordPress/gutenberg/pull/55346/files#r1380041928) with `markNextChangeAsNonPersistent` to keep the undo stack working correctly instead.

Reply:
Yeah, it shouldn't be 'block-library.image.alignmentUpdate', instead something generic like 'blockEditor.align.updateAttributes'.

Reply:
@talldan I really fiddled with naming this via the image block - and forgot where this file is 😂 . 
@ajlende @fabiankaegy  I like this option better than the effect idea because I think other blocks may have this option when they opt into align - _because_ align wide and full are opinionated aligns done by the editor so we should allow blocks to manage the effect of this in a standardized way. We are filtering a core behavior which is very standard to WordPress.

Reply:
> We are filtering a core behavior which is very standard to WordPress.

Adding filters for behaviors as specific as align block supports is not standard for Gutenberg. We have things like `blocks.getBlockAttributes`, `editor.blockEdit`, and `blockEditor.useSetting.before` which handle very high-level or generic settings which apply to all blocks or the editor as a whole.

Something like `editor.setAttributes` where `setAttributes` is defined is more in line with the other hooks that we have. However, I don't think that should be added as `setAttriburtes` is called very frequently and it could lead to performance issues.

The effect solution is used in very many places and doesn't require adding public APIs.